File: 12_복합패턴/Sangmin/mvc_flutter/lib/controller/counter_controller.dart
import 'package:mvc_flutter/controller/counter_controller_interface.dart';
import 'package:mvc_flutter/model/counter_model_interface.dart';

/// 뷰를 대상으로 하는 행동만을 정의한다.
class CounterController implements CounterControllerInterface {
  CounterModelInterface model;

  CounterController(this.model);

  @override
  void increment() {
    model.increase();
  }

  @override
  void decrement() {
    model.decrease();
  }

  @override
  void reset() {
    model.reset();
  }
}

File: 12_복합패턴/Tony/src/MVC/counterController.ts
import { CounterModel } from "./counterModel";
import { CounterView } from "./counterView";

// MVC counter controller
export class CounterController {
  private model: CounterModel;
  private view: CounterView;

  constructor(model: CounterModel) {
    this.model = model;
    this.view = new CounterView(this.model, this);
    this.model.addListener(this.view);
  }

  public increase() {
    this.model.increase();
  }

  public decrease() {
    this.model.decrease();
  }

  public getView() {
    return this.view;
  }
}
